utils/pluggable.hh

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2025-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#pragma once
+
+#include <stdexcept>
+#include <optional>
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/gate.hh>
+#include <seastar/core/shared_ptr.hh>
+
+#include "seastarx.hh"
+#include "utils/phased_barrier.hh"
+#include "utils/on_internal_error.hh"
+
+namespace utils {
+
+/// This class provides an interface for pluggable async services
+template <typename T>
+class pluggable {
+    shared_ptr<T> _service;
+    utils::phased_barrier _phaser;
```

Comment:
PR looks good, but why is simple gate not enough and a phaser is needed? There's not going to be any "phases" AFAICS from next patch